src/v/cluster/tx_gateway_frontend.cc

```diff
@@ -928,52 +932,37 @@ ss::future<add_paritions_tx_reply> tx_gateway_frontend::add_partition_to_tx(
        timeout](tx_gateway_frontend& self) mutable {
           return ss::with_gate(
             self._gate, [request = std::move(request), timeout, &self] {
-                return self.do_add_partition_to_tx(request, timeout);
+                return self.with_stm(
+                  [request = std::move(request), timeout, &self](
+                    checked<ss::shared_ptr<tm_stm>, tx_errc> r) {
+                      if (!r) {
+                          return ss::make_ready_future<add_paritions_tx_reply>(
+                            make_add_partitions_error_response(
+                              request, r.error()));
+                      }
+                      auto stm = r.value();
+                      return stm->read_lock().then(
+                        [&self, stm, request = std::move(request), timeout](
+                          ss::basic_rwlock<>::holder unit) mutable {
+                            auto tx_id = request.transactional_id;
+                            return with(
+                                     stm,
+                                     tx_id,
+                                     "add_partition_to_tx",
+                                     [&self,
+                                      stm,
+                                      request = std::move(request),
+                                      timeout]() mutable {
+                                         return self.do_add_partition_to_tx(
+                                           stm, std::move(request), timeout);
+                                     })
+                              .finally([u = std::move(unit)] {});
```

Comment:
same here with respect to stm lifetime as above comment

Reply:
discussed offline, by the code composition stm is alive when the stm's gate is open so extending isn't needed; also we pass stm into the fiber by value so extending isn't needed